### PR TITLE
feat: add yurtadm renew certificate command

### LINKED ISSUE #1290: [feature request]improve bootstrap configuration for YurtHub
**What would you like to be added**:
A bootstrap token should be passed to yurthub(--join-token parameter) for bootstrapping. sometime token without expire date is used for convenient,  so security problem maybe happen because the bootstrap token is always exist on the edge node.

so i think we can improve bootstrap configuration for yurthub, a bootstrap file instead of bootstrap token is passed into YurtHub, and bootstrap file will be removed after Yurthub bootstrap completed, so bootstrap token can not be leaked on the edge node.

the work we need do as following:
- YurtHub: 
      add a new paramter --boostrap-file, and use bootstrap file to bootstrap YurtHub, and we can keep the original `--join-token` parameter for several versions.

- Yurtadm: 
      - Yurtadm should create a bootstrap file for YurtHub boostrapping, and remove this file after YurtHub bootstrapping completed.
      - add a new sub command for Yurtadm to create a temp bootstrap file for YurtHub when YurtHub need to use bootstrap file to update certificate if certificate expired. on the other hand, the temp bootstrap file also should be removed after certificate is updated.

**Why is this needed**:
A bootstrap token should be passed to yurthub(--join-token parameter) for bootstrapping. sometime token without expire date is used for convenient,  so security problem maybe happen because the bootstrap token is always exist on the edge node.

**others**
/kind feature

### PR DESCRIPTION
**Pull request by YTGhost**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
1. add yurtadm renew certificate command
2. gather the yurthub-related functions into `pkg/yurtadm/util/yurthub/yurthub.go`
3. gather `CheckKubeletStatus` function into `pkg/yurtadm/util/kubernetes/kubernetes.go`
4. When building a test cluster, yurthub is also started with a bootstrap file

#### Which issue(s) this PR fixes:

Fixes #1290 

#### Special notes for your reviewer:
/assign @rambohe-ch 

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/yurtadm/constants/constants.go [hunk 1]
```diff
@@ -220,6 +223,7 @@ spec:
     - --server-addr={{.kubernetesServerAddr}}
     - --node-name=$(NODE_NAME)
     - --join-token={{.joinToken}}
```
Comment (rambohe-ch) [inline pkg/yurtadm/constants/constants.go:225]: please don't left `join-token` in Yurthub.yaml.

File: pkg/yurtadm/cmd/renew/certificate/certificate.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	o := newCertificateOptions()
+
+	certificateCmd := &cobra.Command{
+		Use:   "certificate",
+		Short: "Create bootstrap file for yurthub to update certificate",
+		RunE: func(certificateCmd *cobra.Command, args []string) error {
+			if err := o.validate(); err != nil {
+				klog.Fatalf("validate options: %v", err)
+			}
+
```
Comment (rambohe-ch) [inline pkg/yurtadm/cmd/renew/certificate/certificate.go:61]: I think we should verify the certificates of Yurthub expired or not at first. and if the certificate is valid, we can exit `yurtadm renew certificate` directly with some logs.

File: pkg/yurthub/certificate/token/token.go [hunk 3]
```diff
@@ -274,6 +277,9 @@ func (ycm *yurtHubCertManager) getPkiDir() string {
 
 // getBootstrapConfFile returns the path of yurthub bootstrap conf file
 func (ycm *yurtHubCertManager) getBootstrapConfFile() string {
+	if len(ycm.bootstrapFile) != 0 {
+		return ycm.bootstrapFile
+	}
```
Comment (rambohe-ch) [inline pkg/yurthub/certificate/token/token.go:282]: we need to check all codes that invoke `getBootstrapConfFile()` function can work or not if Yurthub restart and bootstrap file doesn't exist. 

The following places can not work:
1. verifyServerAddrOrCleanup()
2. prepareConfigAndCaFile()
3. generateCertClientFn()

File: pkg/yurtadm/constants/constants.go (hunk 1 above)
Comment (YTGhost) [reply pkg/yurtadm/constants/constants.go:225]: Fixed

File: pkg/yurtadm/cmd/renew/certificate/certificate.go (hunk 2 above)
Comment (YTGhost) [reply pkg/yurtadm/cmd/renew/certificate/certificate.go:61]: Fixed

**Comment by YTGhost** [comment]
@rambohe-ch PTAL

**Review by rambohe-ch — APPROVED** [review]

**Comment by rambohe-ch** [comment]
@YTGhost this pull request depends on the #1333 , we need to rebase after #1333 merge.

**Comment by YTGhost** [comment]
@rambohe-ch PTAL

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**